applications/ConstitutiveLawsApplication/tests/cpp_tests/test_small_strain_isotropic_damage_3d.cpp

```diff
@@ -96,201 +102,141 @@ KRATOS_TEST_CASE_IN_SUITE(_ConstitutiveLaw_SmallStrainIsotropicDamage3D, KratosC
+
+    // Simulate the call sequence of the element
+    Vector dummy_vector;
+    cl.InitializeMaterial(material_properties, geometry, dummy_vector);
+
+    for (std::size_t t = 0; t < 10; ++t){
+        for (std::size_t comp = 0; comp < 6; ++comp)
+            strain_vector(comp) = t / 9 * imposed_strain(comp);
+
         if (cl.RequiresInitializeMaterialResponse()){
             cl.InitializeMaterialResponseCauchy(cl_parameters);
         }
+
         cl.CalculateMaterialResponseCauchy(cl_parameters);
+
         if (cl.RequiresFinalizeMaterialResponse()){
             cl.FinalizeMaterialResponseCauchy(cl_parameters);
         }
-        double value;
-
-        // Check damage variable
-        cl.CalculateValue(cl_parameters, DAMAGE_VARIABLE, value);
-        // TODO(marandra): NAN values are not handled correctly by KRATOS CHECK functions
-        if (std::isnan(dvr[t]/value)){
-            KRATOS_CHECK_NEAR(dvr[t], value, tolerance);
-        } else {
-            KRATOS_CHECK_NEAR(dvr[t]/value, 1, tolerance);
-        }
+    }
 
-        // Check strain energy
-        cl.CalculateValue(cl_parameters, STRAIN_ENERGY, value);
-        if (std::isnan(ser[t]/value)){
-            KRATOS_CHECK_NEAR(ser[t], value, tolerance);
-        } else {
-            KRATOS_CHECK_NEAR(ser[t]/value, 1, tolerance);
-        }
+    // Check damage variable and strain energy
+    double value;
+    cl.CalculateValue(cl_parameters, DAMAGE_VARIABLE, value);
+    KRATOS_CHECK_NEAR(ref_damage_variable, value, tolerance);
+    cl.CalculateValue(cl_parameters, STRAIN_ENERGY, value);
+    KRATOS_CHECK_NEAR(ref_strain_energy, value, tolerance);
 
-        // Check stress
-        for (std::size_t comp = 0; comp < 6; ++comp){
-            KRATOS_CHECK_IS_FALSE(std::isnan(stress_vector[comp]));
-            if (std::isnan(stress_vector[comp]/str(t, comp))){
-                KRATOS_CHECK_NEAR(stress_vector[comp], str(t, comp), tolerance);
-            } else {
-                KRATOS_CHECK_NEAR(stress_vector[comp]/str(t, comp), 1, tolerance);
-            }
-        }
-
-        // Check constitutive tensor
-        for (std::size_t i = 0; i < 6; ++i){
-            for (std::size_t j = 0; j < 6; ++j){
-                std::size_t idx = i * 6 + j;
-                KRATOS_CHECK_IS_FALSE(std::isnan(const_matrix(i, j)));
-                if (std::isnan(const_matrix(i, j)/cmr(t, idx))){
-                    KRATOS_CHECK_NEAR(const_matrix(i, j), cmr(t, idx), tolerance);
-                } else {
-                    KRATOS_CHECK_NEAR(const_matrix(i, j)/cmr(t, idx), 1, tolerance);
-                    }
-            }
-        }
-    }
+    // Check stress
+    for (std::size_t comp = 0; comp < 6; ++comp)
+        KRATOS_CHECK_NEAR(stress_vector(comp), ref_stress(comp), tolerance);
 
+    // Check constitutive tensor
+    for (std::size_t i = 0; i < 6; ++i)
+        for (std::size_t j = 0; j < 6; ++j)
+            KRATOS_CHECK_NEAR(const_matrix(i, j), ref_C(j, i), tolerance);
```

Comment:
better use KRATOS_CHECK_VECTOR/MATRIX if possible

Reply:
Didn´t know those. I'm planning a followup PR, updating these same tests for testing behavior under compression, and moving 2D tests from python to cpp. I'll fix the KRATOS_CHECK_MATRIX issue there if you don´t mind.